### ISSUE TITLE
ACLs not preserved on remote backups

### ISSUE THREAD

**Issue by Coolguy3289** [borgbackup/borg#6361]
## Have you checked borgbackup docs, FAQ, and open Github issues?

Yes

## Is this a BUG / ISSUE report or a QUESTION?

BUG/Issue

#### Your borg version (borg -V).
1.2.0 on both local and remote machines

#### Operating system (distribution) and version.
Ubuntu 20.04.4 LTS on both versions
Kernel: 5.4.0-100-generic

#### Hardware / network configuration, and filesystems used.
Both machines are VMs on Proxmox, running ext4 filesystems with ACL support enabled.

#### How much data is handled by borg?
We're only testing right now, so less than a few hundred MB

#### Full borg commandline that lead to the problem (leave away excludes and passwords)
* Remote Machine (Backup Server):
`borg init -e repokey /root/repo`
* Local Machine (Backup Client)
`borg create root@filesrv1:/root/repo::test /storage/netfiles/testfolder`
* Remote Machine after backup completes
`borg mount /root/repo::test /tmp/repotest`
`borg extract /root/repo::test`

## Describe the problem you're observing.
When backing up to a remote machine, ACLs are not preserved. Owner/Group IDs are showing properly, but when running the `getfacl` command on a folder and file that has ACLs set on the client machine, they do not show on the remote backup server where the repos are located. Both mounting and extracting show these symptoms.
#### Can you reproduce the problem? If so, describe how. If not, describe troubleshooting steps you took before opening the issue.
Yes, follow the steps above first. Once the backup has been mounted and extracted (Check the mount then unmount if you wish, it makes no difference), run the following to confirm the issue:
```
cd <mount or extracted dir>
getfacl .
```
You'll note that only the system default ACLs are shown.

**Comment by ThomasWaldmann** [comment]
Try to reproduce:
- using a local repository
- using borg 1.1.x
- use same machine for create and extract
- write a reproducer script that is **as simple as possible** and shows the issue

borg mount is not expected to show ACLs, that is just not implemented (and also documented IIRC) and maybe not even possible with current FUSE api. Thus, please remove anything related to `borg mount`. The only TODO here is to check the docs, if that limitation is not documented clear enough.

borg create should create archives containing ACLs.
borg extract should extract ACLs from an archive that contains ACLs.

**Comment by CompuRoot** [comment]
What I usually do:

1. Install `acl` 
2. `getfacl -R /path/to/backup/dir/before/snapshoting  >perms.acl`
3. do backup

and to  restore permissions: `setfacl --restore=perms.acl`

**Comment by ThomasWaldmann** [comment]
@Coolguy3289 did you see my comment?

I really need this reproduced in a more simpler setup, so it will become clear that the problem is located in borg and not in your setup.

Also, please provide a script **as simple as possible** that reproduces and shows the issue.

**Comment by Coolguy3289** [comment]
> * using a local repository
Did this, ACLs were preserved on the local host.
> * using borg 1.1.x
Did this, Same result to a remote server.
> * use same machine for create and extract
I did this in combination with the local repository suggestion above.

> * write a reproducer script that is **as simple as possible** and shows the issue

I'm not sure what I can put into a script, the commands to run were listed in the original post and deal with multiple servers to demonstrate the issue, the only exception is running `getfacl .`, and that was mentioned later in the original post.

Put simply, this isn't really a *simple* issue to setup for reproduction, you're dealing with multiple servers. Please elaborate if I misunderstood

**Comment by ThomasWaldmann** [comment]
All ACL related code runs client-side and I don't see any chance that it might require a remote repo to demonstrate an ACL issue in borg.

So, I'ld bet your problem is not inside borg, but related to you using a different system setup for extraction.

So, to prove that there is a bug in borg, **demonstrate that borg does not correctly extract ACLs when the system is set up in a way supporting this. The easiest way to do this is to use the same system for extraction.**

When you do that, I would also bet it does not make a difference whether the repo is stored locally or remotely.

For easier reproduction, I need a reproduction script working with a local repo and with same system for archive creation and extraction.

Please reopen this ticket if you can reproduce. If you can't, try to find what problem in your system setup causes the issue.
